### PR TITLE
cmd/snap: handle missing snap-confine

### PR DESCRIPTION
**Pull request by chipaca**

### CODE REVIEW

File: cmd/snap/cmd_run.go [hunk 1]
```diff
@@ -216,13 +217,20 @@ func snapRunHook(snapName, snapRevision, hookName string) error {
 }
 
 func runSnapConfine(info *snap.Info, securityTag, snapApp, command, hook string, args []string) error {
+	snapConfine := filepath.Join(dirs.DistroLibExecDir, "snap-confine")
+	if !osutil.FileExists(snapConfine) {
+		if hook != "" {
+			logger.Noticef("WARNING: skipping runnning snap/hook %s/%s: missing snap-confine", info.Name(), hook)
```
Comment (pedronis) [inline cmd/snap/cmd_run.go]: hook %q of snap %q is more typical

File: cmd/snap/cmd_run.go [hunk 2]
```diff
@@ -216,13 +217,20 @@ func snapRunHook(snapName, snapRevision, hookName string) error {
...
+			return nil
+		}
+		return fmt.Errorf(i18n.G("missing snap-confine: is your snapd too old?"))
```
Comment (pedronis) [inline cmd/snap/cmd_run.go]: snapd package? system snapd?

Comment (niemeyer) [inline cmd/snap/cmd_run.go]: Yeah, "snapd package" would be more clear.

**Comment by zyga** [comment]
I'll tweak the message and land

File: cmd/snap/cmd_run.go (hunk 2 above)
Comment (niemeyer) [inline cmd/snap/cmd_run.go]: Perhaps we can be more prescriptive:

"missing snap-confine: try updating the snapd package"

**Pull request merged**